app/controllers/female_allocations_controller.rb

```diff
@@ -111,4 +114,9 @@ def nomis_offender_id_from_url
   def staff_id
     params.require(:staff_id).to_i
   end
+
+  def save_to_session(key, record)
+    # Store the ActiveRecord object's attributes
+    session[key] = record.attributes
+  end
```

Comment:
I'm not sure about the reason for this method - it doesn't abstract the session concept, so the code still has loading from session and deleting session keys inline.